src/Compilers/CSharp/Portable/Syntax/SyntaxEquivalence.cs

```diff
@@ -129,7 +129,7 @@ private static bool AreEquivalentRecursive(GreenNode? before, GreenNode? after,
                 {
                     case SyntaxKind.Block:
                     case SyntaxKind.ArrowExpressionClause:
-                        return AreNullableDirectivesEquivalent(before, after, ignoreChildNode);
+                        return true;
                 }
```

Comment:
Probably good to doc. From our offline talk I think this is correct. But it's subtle, so docs would be good for the future.